src/vs/platform/telemetry/common/telemetryUtils.ts

```diff
@@ -42,6 +43,27 @@ export function combinedAppender(...appenders: ITelemetryAppender[]): ITelemetry
 
 export const NullAppender: ITelemetryAppender = { log: () => null, dispose: () => TPromise.as(null) };
 
+
+export class LogAppender implements ITelemetryAppender {
+
+	private commonProperties = ['sessionID', 'version', 'timestamp'];
```

Comment:
use `Map`.

Reply:
@jrieken What goes into the individual log files for processes (and so the output channel) is filtered. But the dedicated file `telemetry.log` will contain the complete data.